src/pages/PageRouter.tsx

```diff
@@ -0,0 +1,56 @@
+import { Navigate, Route, BrowserRouter, Routes } from 'react-router-dom'
+
+import Home from './Home'
+import Todo from './Todo'
+import Signin from './Signin'
+import Signup from './Signup'
+
+interface RouteProps {
+  isAuth: string | null
+  children: JSX.Element
+}
+
+function ProtectedRoute({ isAuth, children }: RouteProps) {
+  return isAuth ? children : <Navigate to="/signin" replace />
+}
+function PublicRoute({ isAuth, children }: RouteProps) {
+  return isAuth ? <Navigate to="/todo" replace /> : children
+}
```

Comment:
```suggestion
function ProtectedRoute({ isAuth, fallback, children }: RouteProps) {
  return isAuth ? children : <Navigate to={fallback} replace />
}
function PublicRoute({ isAuth, fallback, children }: RouteProps) {
  return isAuth ? <Navigate to={fallback} replace /> : children
}
```

저는 페이지 내부에서 auth유무로 리다이렉션하는 방법만 생각하고 있었는데,

민지님 덕분에 새로운 설계를 보게 되어서 반갑고 또 감사합니다.

민지님의 좋은 아이디어에 여기서 한 가지 더 의견을 보태고 싶어요.
`fallback` 프롭을 추가하는 건 어떨까요?

 만약 auth조건에 맞지 않는 경우에 리다이렉션할 경로를 하위 Navigate에 넘겨주는 방식으로 수정하면, ProtectedRoute가 조금 더 유연한 컴포넌트가 될 것 같아요.

Reply:
헉 좋은 아이디어인 것 같습니다 👍
그러면 재사용할 수도 있고 좋네요 !!

바꾸고 다시 댓글 달아두겠습니다 ~!

Reply:
수정했습니다 !

Reply:
넵 확인했습니다! 그런데 민지님, isAuth는 왜 타입이 string|null인지 추가로 질문이 생겼는데 여쭤봐도 될까요?ㅎㅎㅎ

Reply:
localStorage.getItem()의 리턴값이 string | null이라서 그렇게 지정해두었습니다 !

처음엔 isAuth = localStorage.getItem() ? true : false로 하고 타입을 boolean으로 바꿀까도 생각해봤는데, 불필요한 과정인 것 같아서 리턴값 타입 그대로 적어두었어요 !

Reply:
아하~ 이해했습니다. 로컬스토리지의 키값이 없으면 null을 반환해서 타입을 위처럼 지정해주신거군요. 저는 isAuth면 boolean타입이지 않을까 해서 여쭤봤습니다!

Reply:
나중에 리팩토링을 진행할 때 로컬스토리지 사용하는 반복로직도 분리하면 좋겠네요! Approve~ 하겠습니다!